scripts/plot.py

```diff
@@ -0,0 +1,337 @@
+
+    output = run_ratrunner(remyccfilename, parameters, **kwargs)
+
+    if "console_file" in kwargs:
+        kwargs["console_file"].close()
+
+    return parse_ratrunner_output(output)
+
+def add_plot(axes, link_speeds, norm_scores, **kwargs):
+    """Adds a plot for the given link-packets-per-ms `link_ppts` and normalized
+    scores `norm_scores` to the `axes`."""
+    return plt.semilogx(link_speeds, norm_scores, axes=axes, **kwargs)
+
+def generate_data_and_plot(remyccfilename, link_ppt_range, parameters, console_dir=None, data_dir=None, axes=None):
+    """For a given RemyCC `remyccfilename`, runs rat-runner once on each link
+    speed in `link_ppt_range` (each being specified in packets per millisecond),
+    with other parameters as specified in the dict `parameters`. Returns the
+    link rates specified under "prior assumptions".
+
+    If `console_dir` is specified, the outputs of each rat-runner run are stored in
+        a file (each) in that directory.
+    If `data_dir` is specified, data from the run is written to a (single) file in
+        that directory.
+    If `axes` is specified, a plot will be added to those axes.
+    """
+    if data_dir:
+        data_filename = "data-{remycc}.csv".format(
+                remycc=os.path.basename(remyccfilename))
+        data_file = open(os.path.join(data_dir, data_filename), "w")
+        data_csv = csv.writer(data_file)
+
+    norm_scores = []
+    npoints = len(link_ppt_range)
+
+    for i, link_ppt in enumerate(link_ppt_range, start=1):
+        parameters["link_ppt"] = link_ppt
+        print("\033[KGenerating score for if={:s}, link={:f} ({:d} of {:d})...".format(
+                    remyccfilename, link_ppt, i, npoints),
+                    file=sys.stderr, end='\r', flush=True)
+        norm_score, sender_data, link_ppt_prior = compute_normalized_score(remyccfilename, parameters, console_dir)
+        norm_scores.append(norm_score)
+        sender_numbers = chain(*sender_data)
+        if data_dir:
+            data_csv.writerow([link_ppt, norm_score] + list(sender_numbers))
+
+    if axes:
+        print("\033[KPlotting for file {}...".format(remyccfilename), file=sys.stderr, end='\r', flush=True)
+        link_speeds = [LINK_PPT_TO_MBPS_CONVERSION*l for l in link_ppt_range]
+        add_plot(axes, link_speeds, norm_scores, label=remyccfilename)
+
+    data_file.close()
+
+    print("\033[KDone file {}.".format(remyccfilename), file=sys.stderr)
+    sys.stderr.flush()
+
+    return link_ppt_prior
+
+def plot_from_original_file(datafilename, axes):
+    """Plots data from the file `datafile` to the axes `axes`."""
+    link_speeds = []
+    norm_scores = []
+    try:
+        datafile = open(datafilename)
+        for line in datafile:
+            row = line.split() # at whitespace, treat consecutive spaces as one
+            row = [float(x) for x in row]
+            link_speeds.append(row[0])
+            norm_score = log2(row[1]/row[0]) - log2(row[2]/150)
```

Comment:
I believe that rat-runner uses natural log, instead of log2 for normalized scores. To be consistent, we should probably also use natural log here.